ci/requirements_files.py

```diff
@@ -0,0 +1,13 @@
+"""Determine the Python requirements files."""
+
+from pathlib import Path
+
+def requirements_files() -> list[str]:
+    """Return the Python requirements files in the requirements directory."""
+    # We never return the internal requirements file, because it does not need to be checked
+    requirements_filenames = ("requirements/requirements.txt", "requirements/requirements-dev.txt")
+    return [f"-r {filename}" for filename in requirements_filenames if Path(filename).exists()]
```

Comment:
Aren't we assuming in `pip-base.sh` that `requirements-dev.txt` exists?

Reply:
It feels like there's quite some duplication indeed, but I'm not sure if that's easy to solve in the bash files

Reply:
I've generalised `ci/requirements_files.py` a little bit so that it takes a template parameter:

```python
"""Determine the Python requirements files.

The script returns the requirements files as space separated string:

$ requirements_files.py
requirements/requirements.txt requirements/requirements-dev.txt

The script takes an optional template argument that is used to wrap each requirements filename. For example:

$ requirements_files.py "-r %s"
-r requirements/requirements.txt -r requirements/requirements-dev.txt
"""
```

Now both `pip_compile` and `pip_audit` can use it:

```bash
run_pip_compile() {
    for requirements_file in $(python $(script_dir)/requirements_files.py); do
        extra=$([[ "$requirements_file" == *"-dev"* ]] && echo "--extra dev" || echo "")
        run pip-compile $extra --output-file $requirements_file pyproject.toml
    done
}

run_pip_audit() {
    run_pipx pip-audit --strict --progress-spinner=off $(python $(script_dir)/requirements_files.py "-r %s")
}
```